src/generator/AutoRest.AzureResourceSchema/ResourceSchemaParser.cs

```diff
@@ -82,6 +82,7 @@ public static class ResourceSchemaParser
                     JsonSchema resourceDefinition = new JsonSchema();
                     resourceDefinition.JsonType = "object";
 
+                    resourceDefinition.ResourceType = resourceType;
```

Comment:
Why not just look for the "type" property instead of creating a ResourceType property?